subsys/bluetooth/mesh/Kconfig

```diff
@@ -51,7 +51,7 @@ endif
 config BT_MESH_NLC_PERF_CONF
 	bool "Enable NLC performance configurations"
 	select BT_MESH_COMP_PAGE_2
-	select BT_MESH_MODEL_EXTENSIONS
+	imply BT_MESH_MODEL_EXTENSIONS
```

Comment:
Hm. If this is not a "this is always required for any node claiming NLC compliance" (which it doesn't look like, seeing as it is made optional now with the `imply`), I don't think it should be here at all, but in the individual samples that want to use it. I general, I like KConfig to be explicit and not magical with options being `imply`ed all over the place, it makes it hard to track what options are actually enabled or disabled after all the config is parsed.

Reply:
If we want to `imply` this anywhere, it should be under `BT_MESH_COMP_PAGE_1` in Zephyr. Not here.

Reply:
Hmm, makes sense to me.

Reply:
> Hmm, makes sense to me.

This:
> I general, I like KConfig to be explicit and not magical with options being implyed all over the place, it makes it hard to track what options are actually enabled or disabled after all the config is parsed.

Reply:
Adding `BT_MESH_MODEL_EXTENSIONS=y` to the relevant samples then. Should it be set to `n` by default? (since its value is checked in BT_MESH_MODEL_EXTENSION_LIST_SIZE)